norse/torch/functional/lif.py

```diff
@@ -111,51 +167,33 @@ def lif_step(
         p (LIFParameters): parameters of a leaky integrate and fire neuron
         dt (float): Integration timestep to use
     """
-    # compute voltage updates
-    dv = dt * p.tau_mem_inv * ((p.v_leak - state.v) + state.i)
-    v_decayed = state.v + dv
-
-    # compute current updates
-    di = -dt * p.tau_syn_inv * state.i
-    i_decayed = state.i + di
-
-    # compute new spikes
-    z_new = threshold(v_decayed - p.v_th, p.method, p.alpha)
-    # compute reset
-    v_new = (1 - z_new) * v_decayed + z_new * p.v_reset
-    # compute current jumps
-    i_new = (
-        i_decayed
-        + torch.nn.functional.linear(input_tensor, input_weights)
-        + torch.nn.functional.linear(state.z, recurrent_weights)
-    )
-
-    return z_new, LIFState(z_new, v_new, i_new)
-
-
-class LIFParametersJIT(NamedTuple):
-    """Parametrization of a LIF neuron
-
-    Parameters:
-        tau_syn_inv (torch.Tensor): inverse synaptic time
-                                    constant (:math:`1/\\tau_\\text{syn}`) in 1/ms
-        tau_mem_inv (torch.Tensor): inverse membrane time
-                                    constant (:math:`1/\\tau_\\text{mem}`) in 1/ms
-        v_leak (torch.Tensor): leak potential in mV
-        v_th (torch.Tensor): threshold potential in mV
-        v_reset (torch.Tensor): reset potential in mV
-        method (str): method to determine the spike threshold
-                      (relevant for surrogate gradients)
-        alpha (torch.Tensor): hyper parameter to use in surrogate gradient computation
-    """
-
-    tau_syn_inv: torch.Tensor
-    tau_mem_inv: torch.Tensor
-    v_leak: torch.Tensor
-    v_th: torch.Tensor
-    v_reset: torch.Tensor
-    method: str
-    alpha: torch.Tensor
+    if p.method == "super" and getattr(norse, "IS_OPS_LOADED"):
+        # Order: [v_leak, v_th, v_reset, tau_mem_inv, tau_syn_inv, alpha]
+        cpp_params = (
+            p.v_leak,
+            p.v_th,
+            p.v_reset,
+            p.tau_mem_inv,
+            p.tau_syn_inv,
+            torch.as_tensor(p.alpha),
+        )
+        z, v, i = torch.ops.norse_op.lif_super_step(
+            input_tensor, state, input_weights, recurrent_weights, cpp_params, dt
+        )
+        return z, LIFState(z=z, v=v, i=i)
+    else:
+        jit_params = LIFParametersJIT(
+            tau_syn_inv=p.tau_syn_inv,
+            tau_mem_inv=p.tau_mem_inv,
+            v_leak=p.v_leak,
+            v_th=p.v_th,
+            v_reset=p.v_reset,
+            method=p.method,
+            alpha=torch.as_tensor(p.alpha),
+        )
+        return _lif_step_jit(
```

Comment:
The only argument for having a JIT version next to the CPP version is that it *should* run when there are no binaries available (for instance on Windows?). But I haven't tested this.